pkg/operator/controller/ingress/controller_test.go

```diff
@@ -1736,8 +1807,13 @@ func Test_computeUpdatedInfraFromService(t *testing.T) {
 				ingressLBs := service.Status.LoadBalancer.Ingress
 				for _, ingress := range ingressLBs {
 					if len(ingress.IP) > 0 {
-						if !slices.Contains(infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs, configv1.IP(ingress.IP)) {
-							t.Errorf("expected Infra CR to contain %s", ingress.IP)
+						if !reflect.DeepEqual(infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs, tc.expectedLBIPs) {
+							t.Errorf("expected Infra CR to contain %s but found %s", tc.expectedLBIPs, infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs)
+						}
+					}
+					if len(ingress.Hostname) > 0 {
+						if !reflect.DeepEqual(infraConfig.Status.PlatformStatus.AWS.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs, tc.expectedLBIPs) {
+							t.Errorf("expected Infra CR to contain %s but found %s", tc.expectedLBIPs, infraConfig.Status.PlatformStatus.AWS.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs)
 						}
```

Comment:
Instead of using the `status.loadBalancer.ingress.[ip|hostname]` to determine the platform, consider just using the platform provided in the test case. When we expand this to azure, this will have to be refactored because both GCP and Azure use `ingress.IP`.

Also, this code is somewhat confusing. We loop through each of the `status.loadBalancer.ingress`, but then in the `reflect.DeepEqual` we don't use the ingress variable. I don't think we need to check `if len(ingress.IP) > 0 {`, we can let the test compare if they are both `nil` too for better coverage (making sure it doesn't add an unwanted IP).

```suggestion
			if updated {
				switch tc.platform.Type {
				case configv1.AWSPlatformType:
					if !reflect.DeepEqual(infraConfig.Status.PlatformStatus.AWS.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs, tc.expectedLBIPs) {
						t.Errorf("expected Infra CR to contain %s but found %s", tc.expectedLBIPs, infraConfig.Status.PlatformStatus.AWS.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs)
					}
				case configv1.GCPPlatformType:
					if !reflect.DeepEqual(infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs, tc.expectedLBIPs) {
						t.Errorf("expected Infra CR to contain %s but found %s", tc.expectedLBIPs, infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs)
					}
				}
			}
```

Let me know if that makes sense.

Reply:
Yes to both. Updated.